be/src/olap/delete_handler.cpp

```diff
@@ -203,7 +205,20 @@ Status DeleteHandler::check_condition_valid(const TabletSchema& schema, const TC
     return Status::OK();
 }
 
-bool DeleteHandler::_parse_condition(const std::string& condition_str, TCondition* condition) {
+Status DeleteHandler::_parse_condition(const DeleteSubPredicatePB& sub_cond,
```

Comment:
1. compaction ---> read delete predicates --> v1
2. read query ---> read delete predicates ---> v2
3. read old delete predicates ---> convert v1 to v2